plugin/unimpaired.vim

```diff
@@ -125,8 +125,32 @@ function! s:fnameescape(file) abort
   endif
 endfunction
 
-nnoremap <silent> <Plug>unimpairedDirectoryNext     :<C-U>edit <C-R>=<SID>fnameescape(fnamemodify(<SID>FileByOffset(v:count1), ':.'))<CR><CR>
-nnoremap <silent> <Plug>unimpairedDirectoryPrevious :<C-U>edit <C-R>=<SID>fnameescape(fnamemodify(<SID>FileByOffset(-v:count1), ':.'))<CR><CR>
+function! s:PreviousFileEntry(count) abort
+  let window = getwininfo(win_getid())[0]
```

Comment:
I swapped to do this instead of having a check against filetype first, d3c4712

I'm unsure what is best or most performant.

Reply:
Performance of an extra equality check isn't worth a second thought, but this needs to be done in a way that handles both a missing `getwininfo()`, and a `getwinfo()` that doesn't have `quickfix` or `loclist`. The new behavior doesn't need to be supported; just don't blow up.

Reply:
We should probably avoid `&filetype` since a user could screw that up, perhaps even deliberately, but it would be a fine plan B for no `getwininfo()` if you're feeling ambitious. If not, ignore it.

Reply:
Added safeguards against that in 1fc86e6

Reply:
`get()` can handle that without `&&`.

There's still no check for `exists('*bufwinfo')`.

Reply:
`get()` used in 8328857. 

Should I check for `*bufwinfo` or is `*getwininfo` sufficient? That should cover `win_getid` too.

Reply:
I meant `getwininfo` yeah. And thanks for checking if it covers both because that was going to be my next question.

Reply:
That's why I prefaced that comment with "should". I'm not entirely sure where to look up if vim cam come with getwininfo() and not win_getid() - but, we can always be extra sure and perform two checks with exists()

Reply:
That is done in bc5f8f0

Reply:
And finally in e70ff1a so it all looks and behaves the same now. And to follow s:fnameescape I extracted s:get_window in 8be5f11 to handle the exists() dance.